src/fireedge/src/server/index.js

```diff
@@ -14,145 +14,141 @@
+import { messageTerminal } from './utils/general'
+import { getFireedgeConfig } from './utils/yml'
+
+setDnsResultOrder()
+
 // set paths
 genPathResources()
 
+const appConfig = getFireedgeConfig()
+
 // set fireedge_key
 genFireedgeKey()
 
 // set logger
-initLogger()
-
-// destructure imports
-const unsecureServer = http.createServer
-const secureServer = https.createServer
+initLogger(appConfig.debug_level, appConfig.truncate_max_length)
 
 const app = express()
 const basename = defaultAppName ? `/${defaultAppName}` : ''
 
 let frontPath = 'client'
+let remoteModulesPath = 'modules'
+
+if (process.env.NODE_ENV === defaultWebpackMode) {
+  frontPath = `../../dist/${frontPath}`
+  remoteModulesPath = `../../dist/${remoteModulesPath}`
+}
 
 // settings
-const appConfig = getFireedgeConfig()
 const host = appConfig.host || defaultHost
 const port = appConfig.port || defaultPort
 
-if (env && env.NODE_ENV && env.NODE_ENV === defaultWebpackMode) {
-  try {
-    const webpackConfig = require('../../webpack.config.dev.client')
-    const compiler = webpack(webpackConfig)
-
-    app.use(
-      // eslint-disable-next-line import/no-extraneous-dependencies
-      require('webpack-dev-middleware')(compiler, {
-        publicPath: webpackConfig.output.publicPath
-      })
-    )
-
-    app.use(
-      // eslint-disable-next-line import/no-extraneous-dependencies
-      require('webpack-hot-middleware')(compiler, {
-        log: false,
-        path: '/__webpack_hmr',
-        heartbeat: 10 * 1000
-      })
-    )
-  } catch (error) {
-    if (error) {
-      messageTerminal({
-        color: 'red',
-        error
-      })
-    }
-  }
-  frontPath = '../client'
-}
+app.use(helmet.xssFilter())
 app.use(helmet.hidePoweredBy())
 app.use(compression())
 app.use(`${basename}/client`, express.static(resolve(__dirname, frontPath)))
 app.use(`${basename}/client/*`, express.static(resolve(__dirname, frontPath)))
 
+// Remote modules serving
+app.use(
+  `${basename}/modules`,
+  express.static(resolve(__dirname, remoteModulesPath))
+)
+app.use(
+  `${basename}/modules/*`,
+  express.static(resolve(__dirname, remoteModulesPath))
+)
+
 const loggerMiddleware = getLoggerMiddleware()
 if (loggerMiddleware) {
   app.use(loggerMiddleware)
 }
 
-// cors
+// CORS
 if (appConfig.cors) {
   app.use(cors())
 }
 // post params parser body
 app.use(express.urlencoded({ extended: false }))
 app.use(express.json())
 
-app.use(`${basename}/api`, entrypointApi) // opennebula Api routes
+app.use(`${basename}/api`, entrypointApi) // OpenNebula Api routes
 const frontApps = Object.keys(defaultApps)
-frontApps.forEach(frontApp => {
+frontApps.forEach((frontApp) => {
   app.get(`${basename}/${frontApp}`, entrypointApp)
   app.get(`${basename}/${frontApp}/*`, entrypointApp)
 })
-app.get('/*', (req, res) => res.redirect(`/${defaultAppName}/provision`))
+app.get('/*', (_, res) => res.redirect(`/${defaultAppName}/sunstone`))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).